File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.cpp
class Solution {
public:
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        int n = l.size();
        vector<bool> ans;
        for(int i = 0; i<n; i++) {
            vector<int> v = vector<int>(nums.begin() + l[i], nums.begin() + r[i] + 1);
            sort(v.begin(), v.end());
            bool flag = true;
            
            if(v.size() > 1) {
                int diff = v[1] - v[0];
                for(int i = 1; i<v.size() - 1; i++) {
                    if(v[i+1] - v[i] != diff) {
                        flag = false;
                        break;
                    }
                }
            }
            else flag = false;
            ans.push_back(flag);
        }
        
        return ans;
    }
};

File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        int n = l.length;
        List<Boolean> ans = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            List<Integer> subarray = new ArrayList<>();
            for (int j = l[i]; j <= r[i]; j++) {
                subarray.add(nums[j]);
            }
            Collections.sort(subarray);
            boolean flag = true;

            if (subarray.size() > 1) {
                int diff = subarray.get(1) - subarray.get(0);
                for (int k = 1; k < subarray.size() - 1; k++) {
                    if (subarray.get(k + 1) - subarray.get(k) != diff) {
                        flag = false;
                        break;
                    }
                }
            } else {
                flag = false;
            }
            ans.add(flag);
        }

        return ans;
    }
}